homeassistant/components/ruckus_unleashed/coordinator.py

```diff
@@ -0,0 +1,35 @@
+"""Ruckus Unleashed DataUpdateCoordinator."""
+from datetime import timedelta
+
+from pyruckus import Ruckus
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import _LOGGER, CLIENTS, DOMAIN, SCAN_INTERVAL
+
+
+class RuckusUnleashedDataUpdateCoordinator(DataUpdateCoordinator):
+    """Coordinator to manage data from Ruckus Unleashed client."""
+
+    def __init__(self, hass: HomeAssistant, *, ruckus: Ruckus):
+        """Initialize global Ruckus Unleashed data updater."""
+        self.ruckus = ruckus
+
+        update_interval = timedelta(seconds=SCAN_INTERVAL)
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=update_interval,
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch Ruckus Unleashed data."""
+        try:
+            return {
+                CLIENTS: await self.hass.async_add_executor_job(self.ruckus.clients)
+            }
+        except Exception as error:  # pylint: disable=broad-except
```

Comment:
You should catch specific exceptions only. We already have a broad catch in the coordinator. Here is the list what we catch: https://github.com/home-assistant/core/blob/8d687c951a32ff4da1cb06c534f24bf40709b33d/homeassistant/helpers/update_coordinator.py#L141-L175

Reply:
Ah ok I'll update that, thank you

Reply:
Tweaked the error handling and now it will rely on built in coordinator error handling a lot more cleanly.